elasticsearch/repo.sls

```diff
@@ -1,10 +1,11 @@
 # Install Elasticsearch repository
 {%- set repo_version = salt['pillar.get']('elasticsearch:repo_version', '1.1') %}
+{%- set repo_url = salt['pillar.get']('elasticsearch:repo_url', "deb http://packages.elastic.co/elasticsearch/{{ repo_version }}/debian stable main") %}
```

Comment:
Also: you cannot use jinja open/close tags within Jinja as this results in:

```
          ID: elasticsearch
    Function: pkg.installed
      Result: False
     Comment: An error was encountered while installing package(s): W: Failed to fetch http://packages.elastic.co/elasticsearch/{{/dists/repo_version/main/binary-amd64/Packages  404  Not Found [IP: 54.243.211.74 80]
              
              W: Failed to fetch http://packages.elastic.co/elasticsearch/{{/dists/repo_version/}}/debian/binary-amd64/Packages  404  Not Found [IP: 54.243.211.74 80]
              
              W: Failed to fetch http://packages.elastic.co/elasticsearch/{{/dists/repo_version/stable/binary-amd64/Packages  404  Not Found [IP: 54.243.211.74 80]
              
              W: Failed to fetch http://packages.elastic.co/elasticsearch/{{/dists/repo_version/main/binary-i386/Packages  404  Not Found [IP: 54.243.211.74 80]
              
              W: Failed to fetch http://packages.elastic.co/elasticsearch/{{/dists/repo_version/}}/debian/binary-i386/Packages  404  Not Found [IP: 54.243.211.74 80]
              
              W: Failed to fetch http://packages.elastic.co/elasticsearch/{{/dists/repo_version/stable/binary-i386/Packages  404  Not Found [IP: 54.243.211.74 80]
              
              E: Some index files failed to download. They have been ignored, or old ones used instead.
     Started: 13:02:34.205927
    Duration: 7111.5 ms
     Changes:   ```

You can concatenate the string with a `~`